engine/item_effects.asm

```diff
@@ -892,10 +892,10 @@ HeavyBallMultiplier:
 
 .WeightsTable:
 ; weight factor, boost
-	db 2048 >> 8, 0
-	db 3072 >> 8, 20
-	db 4096 >> 8, 30
-	db 65280 >> 8, 40
+	db HIGH(2048),   0
+	db HIGH(3072),  20
+	db HIGH(4096),  30
+	db HIGH(65280), 40
```

Comment:
It looks like this function can infinite if the upper half of the weight is 0xff.

Reply:
`HIGH(65280)` = `0xff`, so nothing higher is possible.

Reply:
Right, and it only breaks on less than.